MarkBind/markbind
# Avoid duplicating search results

**What is the purpose of this pull request?**




Resolves #311 
Resolves #1972 

**Overview of changes:**
Add in a check to ensure that headers at the very top of the page which are an exact match of the title are not displayed in the search results. Also changed the design of the results dropdown to avoid duplicating results and to make clear which results are page titles and which are headers within the page.

**Anything you'd like to highlight/discuss:**
Screenshots are in the comments below!

**Testing instructions:**
For the main test of the implementation, create a page with the page title the same as the first and second heading. The first heading should not show up, while the second heading does. 

For testing of heading keywords, create a page with the page title the same as the first heading, and with the first heading having some keyword not contained in heading/page title. Upon searching the keyword, only the page title should turn up (i.e. without the first heading below it).

For testing of UI/UX, use long headings and keywords, with some having no spaces within the words as well.

For page keywords and heading indexing, please see [this comment](https://github.com/MarkBind/markbind/pull/1968#discussion_r939712691) for the issues.



**Proposed commit message: (wrap lines at 72 characters)**
Update displaying of search results

Search results currently show a duplicate column for headings.
Topmost headings are also included in the search results even if
they are exactly the same as the title.

This duplication is redundant and could appear to the user as
being buggy due to the repetition.

To avoid these issue, let's
* remove the duplicate results column for headings
* remove the duplication of search results if the topmost heading
matches the title exactly



---

**Checklist:** :ballot_box_with_check:
